### PR TITLE
Movie making

### PR DESCRIPTION
**Pull request by OrtnerMichael**
# Related Issues
- #766 

The inbuilt export example should be reviewed when the `zoom` kwarg is fixed.

### CODE REVIEW

File: docs/_pages/user_guide/examples/examples_vis_animations.md [hunk 1]
```diff
@@ -84,3 +84,201 @@ with magpy.show_context([magnet, sensor], backend='plotly', animation=True) as s
...
+        frame_files = create_frames(frames, temp_dir)
+        create_gif(frame_files, frame_time, output_file)
+
+if __name__ == "__main__":
+    main()
+```
+
+<img src="../../../_static/videos/example_gif2.gif" width=50% align="center">
+
+Notic that when providing a canvas, no update to its layout is performed by Magpylib, unless explicitly specified by setting `canvas_update=True` in `show()`. By default `canvas_update="auto"` only updates the canvas if is not provided by the user. Details can be found in the [graphics documentation](guide-graphics-canvas).
```
Comment (Alexboiboi) [inline docs/_pages/user_guide/examples/examples_vis_animations.md:193]: Notice

**Comment by Alexboiboi** [comment]
Hi @OrtnerMichael,

The plotly example can also be modified without writing files to disk, the figure exporter can write bytes directly

```python
import magpylib as magpy
from PIL import Image
import io


def create_gif(images, frame_time, output_file):
    """Create GIF from frames in the temporary directory."""
    frames = [Image.open(io.BytesIO(data)) for data in images]
    if frames:
        frames[0].save(
            output_file,
            format="GIF",
            append_images=frames[1:],
            save_all=True,
            duration=frame_time,
            loop=0,  # Infinite loop
        )


def create_frames(frames):
    """Create frames with Pyvista."""

    # Create Magpylib objects
    mag1 = magpy.magnet.CylinderSegment(
        dimension=(3, 4, 1, 0, 45), polarization=(0, 0, 1)
    )
    mag2 = magpy.magnet.CylinderSegment(
        dimension=(2, 3, 1, 0, 45), polarization=(0, 0, -1)
    )
    images = []
    for i in range(frames):
        # Set object position
        mag1.rotate_from_angax(360 / frames, axis="z")
        mag2.rotate_from_angax(-360 / frames, axis="z")

        fig = magpy.show(
            mag1, mag2, return_fig=True, backend="plotly", style_legend_show=False
        )

        # Edit figure in Plotly
        fig.add_scatter3d(
            x=(0, 0, 4, 4, 0),
            y=(0, 0, 0, 0, 0),
            z=(-2, 2, 2, -2, -2),
            mode="lines",
            line_color="black",
        )

        # Customize layout
        fig.update_layout(
            scene=dict(
                camera_eye={"x": 1.5, "y": 1.5, "z": 1.5},
                camera_up={"x": 0, "y": 0, "z": 1},
                xaxis_range=(-5, 5),
                yaxis_range=(-5, 5),
                zaxis_range=(-5, 5),
            ),
            showlegend=False,
            margin=dict(l=0, r=0, t=0, b=0),
        )

        # Create screenshot (requires kaleido package)
        print(f"Writing frame {i+1:3d}/{frames}")
        img = fig.to_image(format="png", width=500, height=500)
        images.append(img)
    return images


def main():
    frames = 50
    frame_time = 50
    output_file = "my.gif"

    images = create_frames(frames)
    create_gif(images, frame_time, output_file)


if __name__ == "__main__":
    main()
```

**Comment by OrtnerMichael** [comment]
@Alexboiboi 

Thanky for the inputs - I think this massively improves the examples.
I have modified Plotly as you suggested and Pyvista following the same thought.

Please review :)

**Comment by Alexboiboi** [comment]
What about matplotlib?

**Comment by OrtnerMichael** [comment]
Because Matplotlib is "a bit limited" in the 3D graphic department - which is the only reason I use the other backends :).

**Comment by OrtnerMichael** [comment]
@Alexboiboi 

can you please review and merge ?

**Review by Alexboiboi — APPROVED** [review]

**Pull request merged**